src/components/tree-detail/stores/tree-store.tsx

```diff
@@ -95,25 +89,17 @@ export const useTreeStore = create<TreeStore>()((set, get) => ({
 		if (!treeId) {
 			return;
 		}
+		const { data, error } = await supabaseClient
+			.rpc("waterings_for_tree", { t_id: treeId })
+			.select("*")
+			.order("id", { ascending: false })
+			.abortSignal(abortController.signal);
 
-		const lastWateredUrl = `${
-			import.meta.env.VITE_API_ENDPOINT
-		}/get/lastwatered?id=${treeId}`;
-
-		const res = await fetch(lastWateredUrl, {
-			method: "GET",
-			headers: {
-				Authorization: `Bearer ${import.meta.env.VITE_ANON_KEY}`,
-				"Content-Type": "application/json",
-			},
-			signal: abortController.signal,
-		});
-
-		if (!res.ok) {
+		if (error) {
 			throw new Error("Failed to fetch tree watering data");
 		}
 
-		const treeWateringData: TreeWateringData[] = (await res.json()).data;
+		const treeWateringData: TreeWateringData[] = data;
```

Comment:
This might also be solved by typing the client

Reply:
yes, you are right. there might be many more occasions where we could use the generated types from the database. I will handle it in a separate PR.